tapir/coop/models.py

```diff
@@ -719,3 +718,63 @@ def populate(
             old_frozen=old_frozen,
             new_frozen=new_frozen,
         )
+
+class ResignedMembership(models.Model):
+    share_owner = models.ForeignKey(
+        ShareOwner, on_delete=models.deletion.CASCADE, verbose_name=_("Shareowner"), related_name="share_owner",
+    )
+    cancellation_date = models.DateField(
+        default=timezone.now,
+        blank=True,
+        )
+    pay_out_day = models.DateField(null=True)
+    cancellation_reason = models.CharField(max_length = 1000)
+    coop_buys_shares_back = models.BooleanField()
+    willing_to_gift_shares_to_coop = models.BooleanField()
+    transfering_shares_to = models.ForeignKey(
+        ShareOwner, on_delete=models.deletion.CASCADE, verbose_name="OwnerToTransfer", null=True, related_name="owner_to_transfer",
+    )
+    paid_out = models.BooleanField(default=False)
+
+    class ResignedMemberQuerySet(models.QuerySet):
+        def with_term(self, search_string: str):
+            searches = [s for s in search_string.split(" ") if s != ""]
+
+            for search in searches:
```

Comment:
Better but not quite there yet!

In your example, `filtered_members` is a QuerySet, where `id_filter` is a [Q-Object](https://docs.djangoproject.com/en/3.2/topics/db/queries/#complex-lookups-with-q-objects), so I don't think you can combine them.
Also, when combining the filters, we want to combine them with OR instead of AND.
You example would become (I think):
```python
 def with_name_or_id(self, search_string: str):
            id_filter = (Q(share_owner__id__icontains=search_string))
            member_filter = Q(share_owner__inShareOwner.objects.with_name(search_string))
            combined_filters = id_filter | member_filter 
            return self.filter(combined_filters)
```

Reply:
Fixed